src/Couchbase.Lite.Tests.Shared/P2PTest.cs

```diff
@@ -410,12 +410,30 @@ private void CollectionPushPullReplication(bool continuous)
                 config.AddCollection(colADb);
 
                 RunReplication(config, 0, 0);
+                bool retry = false;
+                int retryCount = 0;
+                do {
+                    // Do a retry loop because there is some sort of race, but it most likely
+                    // lies in the Mock Server implementation.  The funny part is that this never
+                    // seems to actually loop, but rather its mere presence seems to cause the test
+                    // to consistently pass.  Also note, only seems to have this race issue with
+                    // UWP .NET Native.
+                    retry = false;
+                    try {
+                        // Check docs are replicated between collections colADb & colAOtherDb
+                        colAOtherDb.GetDocument("doc").GetString("str").Should().Be("string");
+                        colAOtherDb.GetDocument("doc1").GetString("str1").Should().Be("string1");
```

Comment:
How do these tests fail.  In Java, they fail by throwing an exception.  If they do that here, the failure will disappear in the retry loop?

Reply:
The assertion fails, which causes an exception.